commit 6afca163c6f9158c8d995916782cd765188cbdf1 (refs/changes/88/3288/3)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2018-02-01T22:30:40+00:00 (1 year, 8 months ago)
    
    OS-6582 libfakekernel changes break SmartOS platform build
    Reviewed by: Dan McDonald <danmcd@joyent.com>
    Approved by: Dan McDonald <danmcd@joyent.com>

diff --git a/usr/src/lib/libzpool/Makefile.com b/usr/src/lib/libzpool/Makefile.com
index e4511dc50a..b5a25f2b62 100644
--- a/usr/src/lib/libzpool/Makefile.com
+++ b/usr/src/lib/libzpool/Makefile.com
@@ -21,7 +21,7 @@
 #
 # Copyright (c) 2005, 2010, Oracle and/or its affiliates. All rights reserved.
 # Copyright (c) 2013, 2016 by Delphix. All rights reserved.
-# Copyright 2017 Joyent, Inc.
+# Copyright 2018 Joyent, Inc.
 #
 
 LIBRARY= libzpool.a
@@ -75,6 +75,11 @@ CPPFLAGS +=	$(INCS)	-DDEBUG -D_FAKE_KERNEL
 LINTFLAGS +=	-erroff=E_STATIC_UNUSED $(INCS)
 LINTFLAGS64 +=	-erroff=E_STATIC_UNUSED $(INCS)
 
+# The following is needed to fix the SmartOS build; see OS-6582. We cannot do
+# a conditional appendage to INCS, since that breaks the lint build.
+CFLAGS += -isystem $(ROOT)/usr/include
+CFLAGS64 += -isystem $(ROOT)/usr/include
+
 CERRWARN +=	-_gcc=-Wno-parentheses
 CERRWARN +=	-_gcc=-Wno-switch
 CERRWARN +=	-_gcc=-Wno-type-limits
